### ISSUE TITLE
proxy.uku.im: port 8888 closed?

### ISSUE THREAD

**Issue by zmwangx** [Unblocker/Unblock-Youku#531]
Without BS (well, I'll state problem Y first and leave problem X to the end, in case anyone's interested):

```
> nmap -p 8888 proxy.uku.im

Starting Nmap 7.00 ( https://nmap.org ) at 2015-11-30 03:49 PST
Nmap scan report for proxy.uku.im (43.241.217.230)
Host is up (0.24s latency).
PORT     STATE  SERVICE
8888/tcp closed sun-answerbook

Nmap done: 1 IP address (1 host up) scanned in 1.19 seconds
```

Same thing with `curl`:

```
> curl -vx proxy.uku.im:8888 http://www.youku.com/
*   Trying 43.241.217.230...
* connect to 43.241.217.230 port 8888 failed: Connection refused
* Failed to connect to proxy.uku.im port 8888: Connection refused
* Closing connection 0
curl: (7) Failed to connect to proxy.uku.im port 8888: Connection refused
```

I wonder if the server is having trouble again? However, **the Chrome extension seems to be working fine**, so I'm torn and confused now.

I have looked through previous issues like #427 (conclusion there was server reached bandwidth limit) and #453 (server constantly being DOS'ed), but those issues are a few months old now and are not really the same kind of breakage. Any ideas?

---

PS. A bit of background stuff: I was trying to `proxy.uku.im:8888` as an extractor proxy for [`you-get`](https://github.com/soimort/you-get), since for the first time in forever I was seeing Youku/Tudou's region block even on the command line (something seems to have changed in the past month or so?). But as said it didn't work as expected.

Meanwhile, closed sourced server-based downloaders like BigRats still work, but BigRats for one tops out at 超清 for Youku/Tudou (whereas you-get can grab 1080p when available), so it's really a bummer. Any suggestions from the kind folks around here?

By the way, please feel free to reply in Chinese... Just thought writing in English on GitHub might reach more people.

**Comment by yanggis** [comment]
proxy.uku.im allows some SPECIFIC links to pass; but, obviously, the links DO NOT include the link of www.youku.com.

**Comment by zmwangx** [comment]
Ah okay, there was just an example, and it didn't even get to the pass or not pass stage yet... Anyway, if you need another real world example, let's try a random video page (I suppose that's allowed?):

```
curl -vx proxy.uku.im:8888 http://v.youku.com/v_show/id_XMTQwMDAzNzQ1Mg\=\=.html
*   Trying 43.241.217.230...
* connect to 43.241.217.230 port 8888 failed: Connection refused
* Failed to connect to proxy.uku.im port 8888: Connection refused
* Closing connection 0
curl: (7) Failed to connect to proxy.uku.im port 8888: Connection refused
```

**Comment by tommyzhaozbh** [comment]
Well, for the first question as you stated at the beginning, if what I remember is correct, the proxy server sets port 443 for the use of proxy. I can't remember which topic talked about the conflict of the port so the proxy port is set as 443. Maybe port 8888 is closed.
Also the server has a white list to filter what links to proxy. If the link you request is not in the proxy list, the proxy server will refuse your connection request. 
For the question of BigRat, the cs said that the software is only available for free videos download and not suitable for paid vidvideos (including vip services like 1080p) since their server does not have the vip authority to access the 1080p videos. So you cannot download that. 
PS: it is lucky enough that BigRats still support youku as iqiyi has sent letter to them forbidding them from doing so....

**Comment by zmwangx** [comment]
> Well, for the first question as you stated at the beginning, if what I remember is correct, the proxy server sets port 443 for the use of proxy.

Yes 443 it is! At least the curl request gets through now. `you-get` still can't get past the restriction even after the extractor proxy, so I guess something changed in the backend that we can't help... Anyway, that's out of scope for Unblock Youku, so I'll go ahead and close the issue. Thanks.

> For the question of BigRat, the cs said that the software is only available for free videos download and not suitable for paid vidvideos (including vip services like 1080p) since their server does not have the vip authority to access the 1080p videos. So you cannot download that. 

Yeah I know all the legal ramifications :) In fact, ￥20/mo is a very fair price that I would go "shut up and take my money" in a blink... But I'm outside Mainland China so sadly no one's gonna take my money. By the way, `you-get` currently is still able to download Iqiyi videos at 1080p. Not sure how long it will last though.

**Comment by tommyzhaozbh** [comment]
But actually I am quite frustrated about the news that BigRats will not support iqiyi any more. You said that you can still download it. How to do that? Do I need to download or install something?

**Comment by zmwangx** [comment]
[`you-get`](https://github.com/soimort/you-get).

Tons better:

1. Command line, scriptable, can easily spawn loads of tasks with no human intervention;
2. Open source;
3. `you-get` isn't parallel by default, but it's able to extracts segment URLs (`you-get -u`), so you can write your own parallel wrapper, instead of being restricted to 5 concurrent downloads, as is in BigRats' case (well that's not the case for BigRats' website, but I usually go with the client because the website has a lot of restrictions, if memory serves).

**Comment by zmwangx** [comment]
For future reference, the issue I encountered even after enabling the extractor proxy was a bug in you-get's `tudou.py`. See https://github.com/soimort/you-get/issues/787.

**Comment by tommyzhaozbh** [comment]
BTW, I've just tried out the you-get in the python environment and I always encountered the syntax problem when I enter the symbol $. What should I enter actually when I want to execute you-get?
![capture](https://cloud.githubusercontent.com/assets/8757104/11802962/d7140a5e-a330-11e5-8459-a8b034142e45.PNG)

**Comment by zmwangx** [comment]
@tommyzhaozbh `you-get` is a command line program, and hence installs a console script named `you-get` (or `you-get.exe` on Windows, I suppose).

**Comment by tommyzhaozbh** [comment]
but why there is an arrow pointing at the symbol $? Is there any syntax problem?

**Comment by zmwangx** [comment]
What? `$` is a conventional prompt symbol. You shouldn't type that into your terminal. Also, why are you running a command line program in a *python shell*? Use cmd or PowerShell instead.

**Comment by tommyzhaozbh** [comment]
I've opened the cmd. Then what's next? What should I type in? (Sorry for tons of stupid questions since I don't have any knowledge of python before...)

**Comment by zmwangx** [comment]
You hardly need any knowledge of python. Just run `you-get` like any other command line program, which means `you-get blah blah` if `you-get` is already on your `PATH`. I don't know much about Python on Windows so if that doesn't work, you'll need a search engine to find out where Python console scripts are installed under Windows, since I have no idea either.
